config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java

```diff
@@ -192,6 +212,7 @@ public void configure(B http) {
 		AuthenticationManager authenticationManager = http.getSharedObject(AuthenticationManager.class);
 		BasicAuthenticationFilter basicAuthenticationFilter = new BasicAuthenticationFilter(authenticationManager,
 				this.authenticationEntryPoint);
+		basicAuthenticationFilter.setSecurityContextRepository(this.securityContextRepository);
```

Comment:
I'm wondering if it makes more sense to provide no default for `securityContextRepository` in this configure (therefore it defaults to `null`), and only set on the filter if not `null`?

Reply:
Assigning a default value and performing a `null` check in the setter seems like a common strategy that is used elsewhere, so I would stay here with this approach and, perhaps, later align other methods in this class.

Reply:
Hi @evgeniycheban! Sorry for the delay in responding.

> Assigning a default value and performing a `null` check in the setter seems like a common strategy that is used elsewhere

I was thinking specifically of configurers where it seems the pattern is often to let the configured `Filter` contain the default and only override the filter (via the setter) if not null.

Can you share an example in other configurers where you see this strategy used? Or are you referring mostly to other classes (besides configurers) in Spring Security?

Reply:
@sjohnr yes, I was referring to other classes where this pattern is commonly used.

Reply:
@sjohnr I've updated the PR to remove the default value and add a `null` check.

Reply:
Awesome! I’ll work on getting this merged shortly.